server/src/main/java/org/opensearch/action/search/GetAllPitNodesRequest.java

```diff
@@ -21,11 +21,22 @@
  */
 public class GetAllPitNodesRequest extends BaseNodesRequest<GetAllPitNodesRequest> {
 
+    // Security plugin intercepts and sets the response with permitted PIT contexts
+    private GetAllPitNodesResponse getAllPitNodesResponse;
+
     @Inject
     public GetAllPitNodesRequest(DiscoveryNode... concreteNodes) {
         super(concreteNodes);
     }
 
+    public void setGetAllPitNodesResponse(GetAllPitNodesResponse getAllPitNodesResponse) {
+        this.getAllPitNodesResponse = getAllPitNodesResponse;
+    }
+
+    public GetAllPitNodesResponse getGetAllPitNodesResponse() {
+        return getAllPitNodesResponse;
+    }
```

Comment:
can we set `ListPitInfo` instead?

Reply:
We can also set failures as part of response which will be left out if we just set `ListPitInfo`